Exporter: Excel XSLSX Streaming POI API throws exception

## 1) Environment
- PrimeFaces version: 6.2
- Does it work on the newest released PrimeFaces version? NO
- Does it work on the newest sources in GitHub? NO
- Application server + version: Jetty 9
- Affected browsers: ALL

## 2) Expected behavior

When using XLSXSTREAM for streaming API of POI the XLSX should be generated correctly.

## 3) Actual behavior

**ERROR: java.lang.IllegalStateException: Could not auto-size column. Make sure the column was tracked prior to auto-sizing the column.**
```
java.lang.IllegalStateException: Could not auto-size column. Make sure the column was tracked prior to auto-sizing the column.
        at org.apache.poi.xssf.streaming.SXSSFSheet.autoSizeColumn(SXSSFSheet.java:1599)
        at org.apache.poi.xssf.streaming.SXSSFSheet.autoSizeColumn(SXSSFSheet.java:1553)
        at org.primefaces.component.export.ExcelExporter.export(ExcelExporter.java:73)
        at org.primefaces.component.export.DataExporter.processAction(DataExporter.java:138)
        at javax.faces.event.ActionEvent.processListener(ActionEvent.java:96)
        at javax.faces.component.UIComponentBase.broadcast(UIComponentBase.java:839)
        at javax.faces.component.UICommand.broadcast(UICommand.java:315)
        at javax.faces.component.UIData.broadcast(UIData.java:1158)
        at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:870)
        at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:1418)
        at com.sun.faces.lifecycle.InvokeApplicationPhase.execute(InvokeApplicationPhase.java:82)
        at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:100)
        at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:201)
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:670)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:864)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
        at org.primefaces.showcase.filter.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:32)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)
        at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:215)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:531)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:319)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:175)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:133)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:754)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:672)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Cannot get best fit column width on untracked column 0. Either explicitly track the column or track all columns.
        at org.apache.poi.xssf.streaming.AutoSizeColumnTracker.getBestFitColumnWidth(AutoSizeColumnTracker.java:278)
        at org.apache.poi.xssf.streaming.SXSSFSheet.autoSizeColumn(SXSSFSheet.java:1596)
```

## 4) Steps to reproduce

Using the Showcase Basic Exporter example changed XLS to XLSXSTREAM. Attempt to download

## 5) Sample XHTML


```xml
<p:dataExporter type="xlsxstream" target="tbl" fileName="cars" />
```

## 6) Sample bean

N/A

--- a/src/main/java/org/primefaces/component/export/ExcelExporter.java
+++ b/src/main/java/org/primefaces/component/export/ExcelExporter.java
@@ -60,7 +60,7 @@ public class ExcelExporter extends Exporter {
             sheetName = "Sheet";
         }
 
-        Sheet sheet = wb.createSheet(sheetName);
+        Sheet sheet = createSheet(wb, sheetName);
 
         if (preProcessor != null) {
             preProcessor.invoke(context.getELContext(), new Object[]{wb});
@@ -103,7 +103,7 @@ public class ExcelExporter extends Exporter {
                 sheetName = "Sheet" + String.valueOf(i + 1);
             }
 
-            Sheet sheet = wb.createSheet(sheetName);
+            Sheet sheet = createSheet(wb, sheetName);
             applyOptions(wb, table, sheet, options);
             exportTable(context, table, sheet, pageOnly, selectionOnly);
 
@@ -247,6 +247,10 @@ public class ExcelExporter extends Exporter {
 
     protected Workbook createWorkBook() {
         return new HSSFWorkbook();
+    }
+
+    protected Sheet createSheet(Workbook wb, String sheetName) {
+        return wb.createSheet(sheetName);
     }
 
     protected void writeExcelToResponse(ExternalContext externalContext, Workbook generatedExcel, String filename) throws IOException {

--- a/src/main/java/org/primefaces/component/export/ExcelXStreamExporter.java
+++ b/src/main/java/org/primefaces/component/export/ExcelXStreamExporter.java
@@ -19,7 +19,9 @@ import java.io.IOException;
 
 import javax.faces.context.ExternalContext;
 
+import org.apache.poi.ss.usermodel.Sheet;
 import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.streaming.SXSSFSheet;
 import org.apache.poi.xssf.streaming.SXSSFWorkbook;
 
 /**
@@ -42,5 +44,13 @@ public class ExcelXStreamExporter extends ExcelXExporter {
     protected void writeExcelToResponse(ExternalContext externalContext, Workbook generatedExcel, String filename) throws IOException {
         super.writeExcelToResponse(externalContext, generatedExcel, filename);
         ((SXSSFWorkbook) generatedExcel).dispose();
+    }
+
+    @Override
+    protected Sheet createSheet(Workbook wb, String sheetName) {
+        SXSSFWorkbook workbook = (SXSSFWorkbook) wb;
+        SXSSFSheet sheet =  workbook.createSheet(sheetName);
+        sheet.trackAllColumnsForAutoSizing();
+        return sheet;
     }
 }

